doc/SONIC_Test_Ingress_Discard_All_HLD.md

```diff
@@ -0,0 +1,715 @@
+Test objective
+
+Verify packet drops by loop-back filter. Loop-back filter means that route to the host with DST IP of received packet exists on received interface
+
+Packet to trigger drop
+```
+###[ Ethernet ]###
+  dst = [auto]
+  src = [auto]
+  type = 0x800
+###[ IP ]###
+    version = 4
+    ttl = [auto]
+    proto = tcp
+    src = [auto]
+    dst = [known_bgp_neighboar_ip]
+...
+```
+
+Test steps
+- PTF host will send IP packet specifying DST IP of VM host. Port to send is port which IP interface is in VM subnet.
+- When packet reaches SONIC DUT, it should be dropped according to the test objective
+- Verify "RX_DRP" counter increment
+
+#### Test case #6
+Test objective
+
+Verify packet which exceed router interface MTU (for IP packets) drops
+Note: make sure that configured MTU on testbed server and fanout are greater then DUT port MTU
+
+Packet to trigger drop
+```
+###[ Ethernet ]###
+  dst = [auto]
+  src = [auto]
+  type = 0x800
+...
+###[ TCP ]###  
+    sport = [auto]
+    dport = [auto]
+    data = [max_mtu + 1]
+...
+```
+
+Test steps
+- PTF host will send IP packet which exceed router interface MTU
+- When packet reaches SONIC DUT, it should be dropped according to the test objective
+- Verify "RX_DRP" counter increment
+
+#### Test case #7
+Test objective
+
+Verify packet with TTL expired (ttl <= 0) drops
+
+Packet to trigger drop
+```
+###[ Ethernet ]###
+  dst = [auto]
+  src = [auto]
+  type = 0x800
+...
+###[ IP ]###
+    version = 4
+    ttl = 0
+    proto = tcp
+    src = [auto]
+    dst = [auto]
+...
+```
+
+Test steps
+- PTF host will send IP packet with TTL = 0
+- When packet reaches SONIC DUT, it should be dropped according to the test objective
+- Verify "RX_DRP" counter increment
+
+#### Test case #8
+Test objective
+
+Verify non-routable packets discarded at router interface
+Packet list:
+- IGMP v1 v2 v3 membership query
+- IGMP v1 membership report
+- IGMP v2 membership report
+- IGMP v2 leave group
+- IGMP v3 membership report
+
+Test steps
+- PTF host will send IGMP v1 v2 v3 membership query
+- When packet reaches SONIC DUT, it should be dropped according to the test objective
+- Verify "RX_DRP" counter increment
+---
+- PTF host will send IGMP v1 membership report
+- When packet reaches SONIC DUT, it should be dropped according to the test objective
+- Verify "RX_DRP" counter increment
+---
+- PTF host will send IGMP v2 membership report
+- When packet reaches SONIC DUT, it should be dropped according to the test objective
+- Verify "RX_DRP" counter increment
+---
+- PTF host will send IGMP v2 leave group
+- When packet reaches SONIC DUT, it should be dropped according to the test objective
+- Verify "RX_DRP" counter increment
+---
+- PTF host will send IGMP v3 membership report
+- When packet reaches SONIC DUT, it should be dropped according to the test objective
+- Verify "RX_DRP" counter increment
+
+#### Test case #9
+Test objective
+
+Verify packet which is not ip (no ip header available) drops
```

Comment:
Please correct description. It is confusing, the reason for drop is that IP header is missed but not "packet is not ip".

Reply:
Fixed.